### PR TITLE
fix(TDP-3451/TDP-4296): recipe visibility management

### PR DESCRIPTION
**Pull request by ncomont**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-3451
https://jira.talendforge.org/browse/TDP-4296

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/596/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/596/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/597/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/597/)

File: dataprep-webapp/src/app/services/playground/playground-service.js [hunk 1]
```diff
@@ -135,11 +135,14 @@ export default function PlaygroundService($state, $rootScope, $q, $translate, $t
 		StateService.setCurrentSampleType(sampleType);
 		FilterService.initFilters(dataset, preparation);
 		updateGridSelection(dataset, preparation);
-		this.updatePreparationDetails();
+		this.updatePreparationDetails().then(() => {
+			if (state.playground.recipe.current.steps.length > 0) {
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/playground/playground-service.js]: Or
```javascript 
if (state.playground.recipe.current.steps.length) {
````

File: dataprep-webapp/src/app/services/playground/playground-service.js [hunk 2]
```diff
@@ -459,6 +459,7 @@ export default function PlaygroundService($state, $rootScope, $q, $translate, $t
 	 */
 	function appendStep(actions) {
 		startLoader();
+		const actualSteps = state.playground.recipe.current.steps.slice(0);
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/playground/playground-service.js]: `0` is useless here

File: dataprep-webapp/src/app/services/playground/playground-service.js [hunk 3]
```diff
@@ -479,7 +480,13 @@ export default function PlaygroundService($state, $rootScope, $q, $translate, $t
 				HistoryService.addAction(undo, redo);
 			})
 			// hide loading screen
-			.finally(stopLoader);
+			.finally(() => {
+				stopLoader();
+				const len = actualSteps.length;
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/playground/playground-service.js]: `actualStepsLength` is too long? :)

File: dataprep-webapp/src/app/services/playground/playground-service.spec.js [hunk 4]
```diff
@@ -1857,6 +1859,30 @@ describe('Playground Service', () => {
...
+				const actions = [{ action: 'uppercase', parameters: parameters }];
+
+				// given : first action call
+				PlaygroundService.appendStep(actions);
+				stateMock.playground.preparation = preparations[0];
+				$rootScope.$digest();
+				$timeout.flush(300);
+
+				expect(StateService.showRecipe).toHaveBeenCalled();
+				showRecipeSpy.calls.reset();
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/playground/playground-service.spec.js]: `StateService.showRecipe.calls.reset();` does not be enough?

File: dataprep-webapp/src/app/services/playground/playground-service.spec.js [hunk 5]
```diff
@@ -1857,6 +1859,30 @@ describe('Playground Service', () => {
...
+
+				// given : second action call
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/playground/playground-service.spec.js]: Could be another test I guess

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/598/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/598/)

**Pull request merged**